pkg/storage/unified/apistore/store.go

```diff
@@ -227,9 +227,10 @@ func (s *Storage) Delete(
 		}
 	}
 
-	// ?? this was after delete before
-	if err := validateDeletion(ctx, out); err != nil {
-		return err
+	if validateDeletion != nil {
```

Comment:
this would panic from dual write when things were failing -- unrelated to the internal id change, but we should keep it